dev/doc/README.md

```diff
@@ -6,25 +6,30 @@ See the first section of [`INSTALL`](../../INSTALL).  Developers are
 recommended to use a recent OCaml version, which they can get through
 opam or Nix, in particular.
 
-## Building `coqtop`
-The general workflow is to first setup a development environment with
-the correct `configure` settings, then hacking on Coq, make-ing, and testing.
+## Building `coqtop` / `coqc` binaries
 
+The recommended development workflow is to rely on Dune.
+See [`build-system.dune.md`](build-system.dune.md).
 
-This document will use `$JOBS` to refer to the number of parallel jobs one
-is willing to have with `make`.
+```
+$ git clone https://github.com/coq/coq.git
+$ cd coq
+$ make -f Makefile.dune # to get an idea of the available targets
```

Comment:
Maybe we could just recommend setting `COQ_USE_DUNE=1` , this way an accidental `make` call won't go back to the coq_makefile build system.

Reply:
I'm hesitant to recommend it since the legacy build system can still be useful for now, e.g. for calling `make doc_gram_rsts`, which we would appreciate if more devs could use. The good thing about #13617 is that it made the two build systems *more* compatible (you don't have to `make clean` to switch from one to the other). But apparently, there's still an issue with `make test-suite`?

Reply:
Docgram can still be used by `make -f Makefile.make doc_gram_rsts` , tho last time I looked we could port this target, or even have `Makefile.dune` call it if you want.

It is correct that running the test-suite in the regular setup will confuse dune, as the test-suite will output objects to the source directory. In general if you mix the two build setups, expect problems I'd say, they are still too diferent, tho after #14176 they will get a bit closer, but likely the test suite problem may remain for a while.